File: src/musicgo/src/shared/classes/SongViewModel.ts

import { UserViewModel } from './UserViewModel';
import { CategoryViewModel } from './CategoryViewModel';
import { VoteViewModel } from './VoteViewModel';
import { AlbumViewModel } from './AlbumViewModel';

export class SongViewModel{

    public Id: string = '00000000-0000-0000-0000-000000000000';

    public Name: string = "";
    
    public User : UserViewModel | null  = new UserViewModel();

    public Category : CategoryViewModel | null = new CategoryViewModel();

    public Album : AlbumViewModel | null = new AlbumViewModel();  

    public Playbacks : number = 0;
    
    public SongKey : string = "";
    
    public UploadDate :Date = new Date();
    
    public VotesCount:number=0;

    public Votes : VoteViewModel[] = [];

    public SongBytes : any;


}

File: ViewModels/ViewModels/SongViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Web;

namespace ViewModels.ViewModels
{
    public class SongViewModel : BaseViewModel
    {

        public AlbumViewModel Album { get; set; }

        public CategoryViewModel Category { get; set; }

        public string Name { get; set; }

        public int Playbacks { get; set; }

        public string SongKey { get; set; }

        public DateTime UploadDate { get; set; }

        public UserViewModel User { get; set; }

        public List<VoteViewModel> Votes { get; set; }

        public HttpPostedFileBase File { get; set; }

        public byte[] SongBytes { get; set; }

        public int VotesCount { get; set; }
    }
}
